src/boot/efi/meson.build

```diff
@@ -199,6 +199,11 @@ efi_cflags = [
         ]
 )
 
+# On some distros, sd-boot/-stub may trigger some bug somewhere that will cause
+# kernel execution to fail. The cause seems to be purely based on code size and
+# always compiling with at least -O1 will work around that.
```

Comment:
Given how strange this is, maybe link the relevant Github issue for reference here